pkg/sql/logictest/testdata/logic_test/collatedstring

```diff
@@ -368,3 +368,32 @@ CREATE TABLE t46570(c0 BOOL, c1 STRING COLLATE en);
 CREATE INDEX ON t46570(rowid, c1 DESC);
 INSERT INTO t46570(c1, rowid) VALUES('' COLLATE en, 0);
 UPSERT INTO t46570(rowid) VALUES (0), (1)
+
+# Test trailing spaces are truncated for char types.
+subtest regression_50015
+
+query T
+SELECT t FROM ( VALUES
+  ('hello '::char(100) collate "en_US"),
+  ('hello t'::char(100) collate "en_US"),
+  ('hello '::string::char(100) collate "en_US"),
+  ('hello t'::string::char(100) collate "en_US")
+) tbl(t)
+----
+hello
+hello t
+hello
+hello t
+
+statement ok
+CREATE TABLE t50015(id int PRIMARY KEY, a char(100), b char(100) COLLATE en); INSERT INTO t50015 VALUES
+  (1, 'hello', 'hello' COLLATE en),
+  (2, 'hello ', 'hello ' COLLATE en),
+  (3, repeat('hello ', 2), repeat('hello ', 2) COLLATE en)
+
+query ITITI
+SELECT id, a, length(a), b, length(b::string) FROM t50015 ORDER BY id ASC
+----
+1  hello                                                                                                 5   hello        5
```

Comment:
Sorry, I'm still not sure what I'm looking for here. So `a` is padded all the way out to 100, even though its length is 5?

Reply:
nah, just trailing spaces gets deleted.
5 is good, because 'hello 6' is 6.